src/main/scala/com/fulcrumgenomics/vcf/filtration/EndRepairArtifactLikelihoodFilter.scala

```diff
@@ -0,0 +1,212 @@
+import htsjdk.samtools.SamPairUtil.PairOrientation
+import htsjdk.samtools.util.{CoordMath, SequenceUtil}
+import htsjdk.samtools.{SAMRecord, SamPairUtil}
+import htsjdk.variant.variantcontext.Genotype
+import htsjdk.variant.vcf.{VCFFilterHeaderLine, VCFInfoHeaderLine}
+/**
+  * Filter that examines SNVs with the alt allele being A or T to see if they are likely the result
+  * of aberrant end-repair and A-base addition.
+  *
+  * @param distance The distance from the end of the template that defines the region where this
+  *                 artifact may occur
+  * @param pValueThreshold the pvalue at and below which mutations are filtered as being likely
+  *                        caused by the artifact
+  */
+class EndRepairArtifactLikelihoodFilter(val distance: Int = 2,
+                                        pValueThreshold: Option[Double] = None) extends SomaticVariantFilter with LazyLogging {
+  val InfoKey           = "ERAP"
+  val InfoDescription   = "P-Value for the End-Repair Artifact test."
+  val FilterKey         = "EndRepairArtifact"
+  val FilterDescription = "Variant is likely an artifact caused by end-repair and A-base addition."
+
+  override val VcfInfoLines: Traversable[VCFInfoHeaderLine]     = Seq(vcfInfoLine(InfoKey, InfoDescription))
+  override val VcfFilterLines: Traversable[VCFFilterHeaderLine] = Seq(vcfFilterLine(FilterKey, FilterDescription))
+
+  /** Only applies to het SNVs where the alt allele is A or T. */
+  override def appliesTo(gt: Genotype): Boolean = {
+    gt.isHet &&
+      !gt.isHetNonRef &&
+      gt.getAlleles.forall(_.length() == 1) &&
+      gt.getAlleles.filter(_.isNonReference).exists(a => a.getBaseString == "A" || a.getBaseString == "T")
+  }
+
+  /** Calculates the p-value and returns it in the map of annotations. */
+  override def annotations(pileup: Pileup[PileupEntry], gt: Genotype): Map[String,Any] = {
+    if (!appliesTo(gt)) {
+      Map.empty
+    }
+    else {
+      val refAllele = SequenceUtil.upperCase(gt.getAlleles.filter(_.isReference).map(_.getBases()(0)).next())
+      val altAllele = SequenceUtil.upperCase(gt.getAlleles.filter(_.isNonReference).map(_.getBases()(0)).next())
+      annotations(pileup, refAllele, altAllele)
+    }
+  }
+
+  /**
+    * Workhorse method that does the majority of the calculation of the p-value.
+    */
+  private[filtration] def annotations(pileup: Pileup[PileupEntry], refAllele: Byte, altAllele: Byte) = {
+    // First we need to count the artifact congruent and non-congruent reference bases
+    var (refGood, refBad, altGood, altBad) = (0, 0, 0, 0)
+    pileup.baseIterator.filter(_.base == refAllele).foreach { rec =>
+      if (isArtifactCongruent(refAllele, altAllele, rec, pileup.pos)) refBad += 1
+      else refGood += 1
+    }
+
+    val lnFractionCongruent = LnProb.toLogProbability(refBad / (refGood + refBad).toDouble)
+    val lnFractionIncongruent = LnProb.toLogProbability(refGood / (refGood + refBad).toDouble)
```

Comment:
Could also be `LnProb.toLogProbability(refGood) - LnProb.toLogProbability(refGood + refBad)`